### PR TITLE
Fix/siteurl replacement

### PR DESCRIPTION
**Pull request by circlecube**
## Proposed changes

This replaces the `{siteUrl}` in the cta links in both scripts.

I also noticed that the myPluginsTools js class was missing the button for when the entitlement is a free plugin and added that. 

Lastly, I updated the debug data with the updated API shape and latest data from staging hiive.

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] I have viewed my change in a web-browser
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by ramyakrishnai — DISMISSED** [review]

File: components/entitlements/index.js [hunk 1]
```diff
@@ -141,7 +141,10 @@ const Entitlements = ( { methods, constants, ...props } ) => {
 	};
     
     const renderCTAUrl = ( url ) => {
-        return url.replace('{siteUrl}', window.NewfoldRuntime.base_url);
+        if ( ! window.NewfoldRuntime.base_url ) {
+			return url.replace('{siteUrl}', ''); // fallback to site relative url if no base_url is found
+		}
+		return url.replace('{siteUrl}', window.NewfoldRuntime.base_url);
```
Comment (wpscholar) [inline components/entitlements/index.js:147]:
```suggestion
        if ( ! window.NewfoldRuntime.siteUrl ) {
			return url.replace('{siteUrl}', ''); // fallback to site relative url if no siteUrl is found
		}
		return url.replace('{siteUrl}', window.NewfoldRuntime.siteUrl);
```

Comment (wpscholar) [reply components/entitlements/index.js:147]: @circlecube 
The `base_url` would be this: `https://evaninstonemountain.com/index.php`
The `siteUrl` would be this: `https://evaninstonemountain.com`

File: includes/js/myPluginsTools.js [hunk 2]
```diff
@@ -53,23 +53,56 @@ class MyPluginTools {
        
     }
 
+    renderCTAUrl( url ) {
+      if ( ! window.NewfoldRuntime.base_url ) {
+        // fallback to site relative url if no base_url is found
+        return url.replace('{siteUrl}', '');
+      }
+      return url.replace('{siteUrl}', window.NewfoldRuntime.base_url);
```
Comment (wpscholar) [inline includes/js/myPluginsTools.js:61]:
```suggestion
      if ( ! window.NewfoldRuntime.siteUrl ) {
        // fallback to site relative url if no siteUrl is found
        return url.replace('{siteUrl}', '');
      }
      return url.replace('{siteUrl}', window.NewfoldRuntime.siteUrl);
```

**Review by wpscholar — CHANGES_REQUESTED** [review]
@circlecube Also check out these asks, which I think change some of the desired logic in the `myPluginsTools.js` file:

- https://jira.newfold.com/browse/PRESS0-2285
- https://jira.newfold.com/browse/PRESS0-2279

**Review by wpscholar — APPROVED** [review]

**Pull request merged**